cmds/init/bgbuild.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Build commands in backgorund processes. This feature certainly makes a
+// non-busybox shell feel more realtime.
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"runtime"
+	"sort"
+)
+
+// Generally shorter commands (ls, cd, ...) are used more often, so they are
+// compiled first. This avoids maintaining a list for compilation order.
+type cmdSlice []os.FileInfo
+
+func (p cmdSlice) Len() int {
+	return len(p)
+}
+
+func (p cmdSlice) Less(i, j int) bool {
```

Comment:
Once we switch to Go1.8 (just released today!), we can replace these three functions with a single `sort.Slice`.